beacon_node/beacon_chain/src/migrate.rs

```diff
@@ -504,249 +473,312 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> BackgroundMigrator<E, Ho
             .into());
         }
 
-        // The new finalized state must be newer than the previous finalized state.
-        // I think this can happen sometimes currently due to `fork_choice` running in parallel
-        // with itself and sending us notifications out of order.
-        if old_finalized_slot > new_finalized_slot {
-            return Ok(PruningOutcome::OutOfOrderFinalization {
-                old_finalized_checkpoint,
-                new_finalized_checkpoint,
-            });
-        }
+        // TODO(hdiff): if we remove the check of `old_finalized_slot > new_finalized_slot` can we
+        // ensure that a single pruning operation is running at once? If a pruning run is triggered
+        // with an old finalized checkpoint it can derive a stale hdiff set of slots and delete
+        // future ones that are necessary breaking the DB.
```

Comment:
I don't _think_ we have to worry about this, because finalization should never go backwards, and we sequence all migration messages (triggered on finalization) through a FIFO channel.